posts/2021-03-03-jakarta-ee-9-compatibility.adoc

```diff
@@ -0,0 +1,26 @@
+---
+layout: post
+title: Open Liberty beta is Jakarta EE 9 compatible
+categories: blog
+author_picture: https://avatars3.githubusercontent.com/jhanders34
+author_github: https://github.com/jhanders34
+seo-title: Open Liberty beta is Jakarta EE 9 compatible - OpenLiberty.io
+seo-description: Open Liberty beta releases have satisified the Jakarta EE 9 TCK and have been added to the Jakarta EE website.
+blog_description: "Open Liberty beta releases have satisified the Jakarta EE 9 TCK and have been added to the Jakarta EE website"
+open-graph-image: https://openliberty.io/img/twitter_card.jpg
+---
+= Open Liberty beta is Jakarta EE 9 compatible
+Jared Anderson <https://github.com/jhanders34>
+:imagesdir: /
+:url-prefix: https://openliberty.io
+:url-about: /
+//Blank line here is necessary before starting the body of the post.
+
+Late last year, the Jakarta EE community released the next step in the evolution of the Jakarta EE specification.  link:https://jakarta.ee/specifications/platform/9/jakarta-platform-spec-9.html[Version 9.0 of the Jakarta EE specification] focuses on renaming packages and namespaces by changing to jakarta prefixed packages names in the API and to the link:https://jakarta.ee/xml/ns/jakartaee/[jakarta.ee namespace] in the deployment descriptor files.  Additionally, the specification makes optional or removes older functions that have been deprecated or stabilized for many years.
+
+The Open Liberty development team has been working over the last year to implement the Jakarta EE 9 specification as it was being finalized.  In an effort to not have to duplicate the same code with references to different API package names, the team developed an link:https://projects.eclipse.org/projects/technology.transformer[Open Source byte code transformation technology].  This technology allows for the same code to be compiled to be Jakarta EE 8 compatible and then be transformed to work with Jakarta EE 9 by manipulating the byte code to convert class and property references from being prefixed with javax to jakarta.
```

Comment:
This technology allows for the same code to be compiled to be Jakarta EE 8 compatible and then be transformed to work with Jakarta EE 9 by manipulating the byte code to convert class and property references from being prefixed with javax to jakarta.
--->
With this technology, Jakarta EE 8 compatible code is transformed to work with Jakarta EE 9 by manipulating the byte code to convert class and property reference prefixes from `javax` to `jakarta`.

Reply:
Updated